src/line/i-chartBuilder.test.js

```diff
@@ -0,0 +1,400 @@
+
+    //make sure the correct button is selected
+    expect(generateButton).toHaveTextContent("Generate chart")
+
+    //Popups
+    window.alert = jest.fn();
+    const alertMock = jest.spyOn(window, 'alert');
+
+    const user = userEvent.setup()
+
+    await user.type(chartTitle, "Random Chart Title")
+    await user.type(xLabel, "X Label Name :)")
+    await user.type(yLabel, "Y Label Name :)")
+    await user.type(xValue, "22")
+
+    await user.click(generateButton)
+    
+    expect(alertMock).toHaveBeenCalledTimes(1)
+})
+
+
+//CLEARING CHART DATA
+test('All chart values are cleared when Clear Chart Data is pressed', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    const chartTitle = domTesting.getByLabelText(document, "Chart title")
+    const xLabel = domTesting.getByLabelText(document, "X label")
+    const yLabel = domTesting.getByLabelText(document, "Y label")
+
+    const buttons = domTesting.queryAllByRole(document, "button")
+    const clearButton = buttons[1]
+
+    //make sure the correct button is selected
+    expect(clearButton).toHaveTextContent("Clear chart data")
+
+    const user = userEvent.setup()
+
+    await user.type(chartTitle, "Random Chart Title")
+    await user.type(xLabel, "X Label Name :)")
+    await user.type(yLabel, "Y Label Name :)")
+
+    xValues = domTesting.getAllByLabelText(document, "X")
+    yValues = domTesting.getAllByLabelText(document, "Y")
+
+    expect(xLabel).toHaveValue("X Label Name :)")
+    expect(yLabel).toHaveValue("Y Label Name :)")
+    expect(chartTitle).toHaveValue("Random Chart Title")
+
+    await user.click(clearButton)
+
+    expect(xLabel).toHaveValue("")
+    expect(yLabel).toHaveValue("")
+    expect(chartTitle).toHaveValue("")
+})
+
+test('X and Y cell count is reduced to 1 when Clear Chart Data is pressed', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    const chartTitle = domTesting.getByLabelText(document, "Chart title")
+
+    const xLabel = domTesting.getByLabelText(document, "X label")
+    const yLabel = domTesting.getByLabelText(document, "Y label")
+    
+    var xValues = domTesting.getAllByLabelText(document, "X")
+    var yValues = domTesting.getAllByLabelText(document, "Y")
+    const xValue = xValues[0]
+    const yValue = yValues[0]
+
+    const buttons = domTesting.queryAllByRole(document, "button")
+    const clearButton = buttons[1]
+    const addButton = buttons[3];
+    
+    //make sure the correct button is selected
+    expect(addButton).toHaveTextContent("+")
+    expect(clearButton).toHaveTextContent("Clear chart data")
+    expect(xValues).toHaveLength(1)
+    expect(yValues).toHaveLength(1)
+
+    const user = userEvent.setup()
+
+    await user.type(chartTitle, "Random Chart Title")
+    await user.type(xLabel, "X Label Name :)")
+    await user.type(yLabel, "Y Label Name :)")
+    await user.type(xValue, "22")
+    await user.type(yValue, "33")
```

Comment:
I think it would be great if data was entered for all of the X/Y input pairs, providing a stronger assertion that the X/Y input elements that exist after the "Clear chart data" button is pressed will be empty.

Reply:
Could you add this in please?